titan-compiler/coder.lua

```diff
@@ -219,7 +219,8 @@ local function codeif(ctx, node, idx)
             %s
             if(%s) {
                 %s
-            } %s
+            }
+			%s
```

Comment:
@andremm was the `reindent_c` function misbehaving in this case? I thought it handled `} else {` correctly since it checks for both `^}` and `{$` in the same pass. I'd rather fix the reindenter than to have `else` in a separate line. :)

Reply:
> @andremm was the reindent_c function misbehaving in this case? I thought it handled }  else { correctly since it checks for both ^} and {$ in the same pass. I'd rather fix the reindenter than to have else in a separate line. :)

sorry @hishamhm, I thought that applying this patch would be easier than having special cases in the reindenter. So, I applied it because leaving `} %s` generates lines containing the pattern `}\t\t{`, and leaving `"else " .. codestat(ctx, node.elsestat)` without left trim generates lines containing the pattern `else \t{`.
Should we roll back and fix that in the reindenter?

Reply:
Anyway, this has been rolled back after fixing the generation of if statements on PR #40.
Now we do not have `else` in a separate line anymore. :)